iceoryx2-cli/iox2-tunnel/src/main.rs

```diff
@@ -10,92 +10,110 @@
 //
 // SPDX-License-Identifier: Apache-2.0 OR MIT
 
-#[cfg(not(debug_assertions))]
-use human_panic::setup_panic;
-#[cfg(debug_assertions)]
-extern crate better_panic;
+#[cfg(target_os = "freebsd")]
+fn main() {
+    println!("Not supported on this platform");
+}
 
+#[cfg(not(target_os = "freebsd"))]
 mod cli;
 
-use anyhow::Result;
-use clap::Parser;
-use cli::Cli;
-use cli::Transport;
-
-use iceoryx2::prelude::*;
-use iceoryx2_bb_log::error;
-use iceoryx2_bb_log::info;
-use iceoryx2_bb_log::set_log_level_from_env_or;
-use iceoryx2_bb_log::LogLevel;
-
-use iceoryx2_tunnels_zenoh::Scope;
-use iceoryx2_tunnels_zenoh::Tunnel;
-use iceoryx2_tunnels_zenoh::TunnelConfig;
+#[cfg(not(target_os = "freebsd"))]
```

Comment:
Why this hop with the `supported_platform`. Would it be possible to have a `main` for non-supported platforms like the one above and another main for the supported ones.

This would simplify the code a bit.